src/backend/Csrs.Api/ApiGateway/ApiGatewayHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Csrs.Api.Configuration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Csrs.Api.ApiGateway
+{
+    public class ApiGatewayHandler : DelegatingHandler
+    {
+
+        public readonly ApiGatewayOptions _apiGatewayOptions;
+
+        public ApiGatewayHandler(
+            IOptions<ApiGatewayOptions> apiGatewayOptions)
+        {
+            _apiGatewayOptions = apiGatewayOptions.Value;
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(_apiGatewayOptions.BasePath)) return await base.SendAsync(request, cancellationToken);
+
+            if (Uri.TryCreate(CombineUrls(_apiGatewayOptions.BasePath, request.RequestUri.PathAndQuery), UriKind.Absolute, out var path))
+            {
+                //request.Headers.Add("MSCRM.SuppressDuplicateDetection", "false");
+
+                //this is to deal with Dynamics, when the method is POST and payload is empty, 
+                //Dynamics still looking for content-type.
```

Comment:
So that we do not forget:
We can remove this code block Sergey (it is from your work on CSRS-290 that I merged and made to work).
It is needed for FAMS3 to handle a special case/one of the recent changes. We do not have a case like this in CSRS. Hence I believe this adjustment is not needed for CSRS.
I can do the cleanup next time when I am on it again. Thanks!